doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,674 @@
+
+    .. code-block:: go
+
+        type ConnectionPoint struct {
+            Type        string
+            IP          string
+            Prefix      uint32
+            InterfaceID uint16
+        }
+    
+- Index identifiers
+    A policy index is to be embedded in the HBH extension and therefore has to be minimal in size.
+    The size of a policy index is 8 bits, whereas identifiers can be a multiple of this (especially global identifiers).
+    The policy index is thus different to the policy identifier. In order to decode which policies are supported on which interfaces,
+    a mapping is required from policy index to local and global identifiers.
+    This mapping is provided by this map.
+- Identifier descriptions
+    Global identifiers can be found in a global datastore, but local identifiers are specific to an AS.
+    This map maps a local policy identifier to its corresponding description.
+- MPLS
+    Routers need to be aware of the supported policy indices and the corresponding MPLS config they need to apply to packets to
+    enforce the policy in the internal network.
+    Routers periodically fetch this map from the control service.
+    A hash of the MPLS map is maintained, such that routers only have to update if their hash differs from the one at the control service.
+- Remote policy cache
+    When a local policy is queried at a remote AS, the resulting policy description is cached at the requesting AS' FABRID Manager,
+    such that subsequent requests can be served from cache.
+
+.. _fabrid_pcb_dissemination:
+
+PCB dissemination
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The *IndexIdentifierMap* and *SupportedIndicesMap* are included in a (unsigned) detachable extension in the PCBs for an AS.
+Hashes of these maps are maintained in a Signed AS Entry, such that the authenticity of these maps can be verified.
+The detachable extension can also be present in the PCB, i.e. it does not have to be detached in all cases, e.g. if there are only very few policies.
+The default is to include the maps directly inside the PCB.
+If the maps are detached, they can be fetched from the control service of that AS and the received maps can be verified with the hashes.
+To ensure a consistent hash calculation, the key entries of these maps have to be sorted, such that they are accessed in a consistent order.
+
+.. _endhost_policy_selection:
+
+Exposing policies to the endhosts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The path combinator finds the most recent FABRID map per AS among the received segments and subsequently uses this map to find the FABRID
+policies that are available for each interface pair of hops.
+If an endhost decides to query for policies at an AS that does not support FABRID, there is a timeout of a few seconds.
+What should be done after the timeout is up to the application.
+The global policy list is not yet implemented.
```

Comment:
"If an endhost decides to query for policies at an AS that does not support FABRID, there is a timeout of a few seconds. What should be done after the timeout is up to the application. The global policy list is not yet implemented." -> Move to end of paragraph

Reply:
Done